pkg/apis/ceph.rook.io/v1/status.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Based on code from https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go
+
+// SetStatusCondition sets the corresponding condition in conditions to newCondition.
+// conditions must be non-nil.
+// 1. if the condition of the specified type already exists (all fields of the existing condition are updated to
+//    newCondition, LastTransitionTime is set to now if the new status differs from the old status)
+// 2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset, and newCondition is appended)
+func SetStatusCondition(conditions *[]Condition, newCondition Condition) {
+	if conditions == nil {
+		return
+	}
+
+	now := metav1.NewTime(time.Now())
+
+	existingCondition := FindStatusCondition(*conditions, newCondition.Type)
+	if existingCondition == nil {
+		if newCondition.LastTransitionTime.IsZero() {
+			newCondition.LastTransitionTime = now
+			newCondition.LastHeartbeatTime = now
+		}
+		*conditions = append(*conditions, newCondition)
+		return
+	}
+
+	if existingCondition.Status != newCondition.Status {
+		existingCondition.Status = newCondition.Status
+		if !newCondition.LastTransitionTime.IsZero() {
+			existingCondition.LastTransitionTime = newCondition.LastTransitionTime
+		} else {
+			existingCondition.LastTransitionTime = now
+		}
+	}
+
+	existingCondition.Reason = newCondition.Reason
+	existingCondition.Message = newCondition.Message
+	if !newCondition.LastHeartbeatTime.IsZero() {
+		existingCondition.LastHeartbeatTime = newCondition.LastHeartbeatTime
+	} else {
+		existingCondition.LastHeartbeatTime = now
+	}
+}
+
+// RemoveStatusCondition removes the corresponding conditionType from conditions.
+// conditions must be non-nil.
+func RemoveStatusCondition(conditions *[]Condition, conditionType ConditionType) {
```

Comment:
Is this function needed anymore? I don't see where it's called.

Reply:
It's not used, but I don't think it's useless. I originally had the thought to change Rook's `Condition` to `metav1.Condition` to allow us to use the helpers provided in the K8s code [here](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go), but that got rid of `LastHeartbeatTime` which could be a breaking change. 

At the top of this file, I have the comment below:
```
// Based on code from https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go
```

Reply:
Makes sense, but I don't see how that answers the question. If nothing is calling this, it's dead code, right?

Reply:
I don't see it that way. It's a convenience method of the API that is unit tested to ensure it works properly for if we want the functionality in the future, which doesn't seem unreasonable to me.

Reply:
Can you save it in a different branch until it's needed in a future PR? If we don't end up using it, we're maintaining dead code.